spec/docker/image_spec.rb

```diff
@@ -398,6 +409,24 @@
     end
   end
 
+  describe '.save_stream' do
+    let(:image) { 'busybox:latest' }
+    let(:expected) do
+      Docker.connection.get(
+        '/images/get',
+        'names' => URI.encode(image)
+      )
+    end
+    let(:actual) { '' }
+
+    it 'yields each chunk of the image' do
+      Docker::Image.save_stream(image) do |chunk|
+        actual << chunk.to_s
+      end
+      expect(actual).to eq(expected)
```

Comment:
Can we perhaps get a list of all files in the tar and use that for comparison? Would probably be saner and easier to debug than this.